### PR TITLE
cmake: Fix build directory exclusion for real

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

Upon further review, the build directory is never set to build_$arch. Currently, our CMake Presets on Ubuntu only use build_ubuntu. However, we can attempt to be flexible here and simply exclude the build directory configured in CMake.

### Motivation and Context

Want this to really be fixed. Follow-up to #11501.

Discovered after tagging 31.0.0-rc1 that the build directory was not excluded from the source package. Dug a bit deeper and realized that the build directory on CI is always `build_ubuntu` and never `build_$arch`/`/build_${CMAKE_SYSTEM_PROCESSOR}`.

We could also simply say "we only support our provided presets" and hardcode this to "build_ubuntu" as we do in `CMakePresets.json`, but this _should_ use that same value.

### How Has This Been Tested?

Tested locally on WSL.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: cmake/linux/cpackconfig.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ endif()
 
 set(CPACK_SOURCE_PACKAGE_FILE_NAME "obs-studio-${CPACK_PACKAGE_VERSION}-sources")
 set(CPACK_SOURCE_GENERATOR "TGZ")
-set(CPACK_SOURCE_IGNORE_FILES "/.git" "/build_${CMAKE_SYSTEM_PROCESSOR}" "/.ccache" "/.deps")
+set(CPACK_SOURCE_IGNORE_FILES "/.git" "${CMAKE_BINARY_DIR}" "/.ccache" "/.deps")
```
[the remark this answers is not in the document]
Comment (RytoEX) [reply cmake/linux/cpackconfig.cmake:19]: IIRC, CMake converts these paths to absolute paths if they are relative, but ultimately uses absolute paths. I'll go re-read the source later to check, but it worked on my machine.

[the remark this answers is not in the document]
Comment (PatTheMav) [reply cmake/linux/cpackconfig.cmake:19]: Would've been easier if CMake would make this explicit in the documentation, but fair enough. 😄

[the remark this answers is not in the document]
Comment (RytoEX) [reply cmake/linux/cpackconfig.cmake:19]: It basically takes a pattern that is checked against the paths to all source files. The hidden part is that CMake composes the source file paths as absolute paths. The underlying calls to assemble the list of source file paths can make relative path lists instead, but isn't used that way here. Unless they plan to change that, the documentation for `CPACK_SOURCE_IGNORE_FILES` could probably be updated.

**Comment by RytoEX** [comment]
Going to merge this. If this somehow still breaks, I'll handle it then.

**Pull request merged**